GeneralsMD/Code/GameEngine/Source/GameLogic/AI/AIStates.cpp

```diff
@@ -2682,7 +2682,15 @@ StateReturnType AIAttackApproachTargetState::updateInternal()
 		}
 		if( victim->testStatus( OBJECT_STATUS_STEALTHED ) && !victim->testStatus( OBJECT_STATUS_DETECTED ) && !victim->testStatus( OBJECT_STATUS_DISGUISED ) )
 		{
+#if RETAIL_COMPATIBLE_CRC
 			return STATE_FAILURE;	// If obj is stealthed, can no longer approach.
+#else
+			const Bool isTargetingMine = weapon && weapon->getDamageType() == DAMAGE_DISARM &&
+				(victim->isKindOf(KINDOF_MINE) || victim->isKindOf(KINDOF_BOOBY_TRAP) || victim->isKindOf(KINDOF_DEMOTRAP));
```

Comment:
If the damage type is `DAMAGE_DISARM`, do we even need to test for the target kindof? We assume that all object kinds can be approached if they can be disarmed, no?